src/sha3/aux_functions.rs

```diff
@@ -98,28 +127,51 @@ pub mod byte_utils {
         rng.gen_bigint(bits as u64)
     }
 
-    /// XORs byte streams in place using iterators
-    /// * `a`: reference to `Vec<u8>`, will be replaced with result of XOR
-    /// * `b`: immut ref to `Vec<u8>`, dropped after function returns
-    /// * `Remark`: Probable bottleneck unless impl with SIMD.
+    /// Performs in-place XOR operation on two byte streams using iterators.
+    ///
+    /// # Arguments:
+    /// * `a` - A mutable reference to a vector of bytes to be XORed and replaced with the result.
+    /// * `b` - An immutable reference to a vector of bytes.
+    ///
+    /// # Return:
+    /// The result of the XOR operation is stored in the vector referenced by `a`.
+    /// 
+    /// # Remarks:
+    /// This function is a probable bottleneck unless implemented with SIMD.
     pub(crate) fn xor_bytes(a: &mut Vec<u8>, b: &Vec<u8>) {
         assert_eq!(a.len(), b.len());
         a.iter_mut().zip(b.iter()).for_each(|(x1, x2)| *x1 ^= *x2);
     }
 
-    ///`return` A string timestamp of current time and date
-    /// corresponding to locale on local machine
+    /// Returns a string representation of the current local date and time.
+    ///
+    /// # Returns:
+    /// A string timestamp representing the current local time and date corresponding to the locale on the local machine.
```

Comment:
although the rest of the library follows this pattern in many places, the return is actually implicit and this extra notation adds unneeded verbosity